### ISSUE TITLE
OS level keybindings don't work in editable fields

### ISSUE THREAD

**Issue by Gozala** [mozilla/browser.html#137]
Examples would be:

`alt backspace` should delete a word
`command left` should move curret to the begining of the line
`command right` should move curret to the end the line
`command shift right` should select text to the end of line
`command shift left` should select text to the begennig of line
`command c` copy
`command v` paste
`command x` cut

I'm sure there are more, but these are the basics that I keep noticing when trying to use this for actual tasks.

**Comment by gordonbrander** [comment]
See also https://github.com/mozilla/browser.html/issues/66

**Comment by gordonbrander** [comment]
Some of these are more used than others, so we could prioritize.

I would also like to make a distinction between "demoable" and "dogfoodable". I would say these are mid-to-low priority for a demo, but pretty important for daily use.

**Comment by paulrouget** [comment]
This is a gecko issue. I think I know how to solve that.

**Comment by paulrouget** [comment]
(e10s issue)

**Comment by paulrouget** [comment]
Doesn't work either with B2G Desktop + e10s + Gaia.

**Comment by paulrouget** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1154182

**Comment by paulrouget** [comment]
I figured out how to fix that (see bugzilla bug), but the `Cmd+arrow` doesn't appear to some the app code. Probably the keybinding code or some focus issue.

**Comment by paulrouget** [comment]
Yeah, it's because of the `preventDefault && stopPropagation`. @Gozala what would you recommend to make it so `preventDefault && stopPropagation` are only called if the callback is successful? check the bindings return value?

**Comment by paulrouget** [comment]
Fixed in latest build.

**Comment by Gozala** [comment]
> Yeah, it's because of the preventDefault && stopPropagation. @Gozala what would you recommend to make it so preventDefault && stopPropagation are only called if the callback is successful? check the bindings return value?

To be honest I think keybinding code is a mess we need to figure out something better for sure. In the meantime solution with return value is probably a best way to go about it.  Although I would add decorator functions like `KeyBindings.preventDefault(handler)` and `KeyBindings.stopPropagation(handler)` to make dealing with special return values more implicit.

I am also struggling to remember why did I went with stop propagation + prevent default in first place, I'll try to find a change I did it in. I suspect `preventDefault` was not really necessary & my understanding is that is the cause of the issue isn't it ?
